src/components/CardsContainer.tsx

```diff
@@ -1,68 +1,78 @@
+		onChange: (allPointsRes) => {
+			const allUsersPointsSorted = [...(allPointsRes.message as unknown as ListenerRes[])]
+
+			allUsersPointsSorted.sort((a, b) => {
+				const aMessage = a.message as unknown as number
+				const bMessage = b.message as unknown as number
+				return bMessage - aMessage
+			})
+			setSortedUsersPoints(allUsersPointsSorted)
+		},
+	})
 	const usersPointsData = (allUsersStoryPoints?.message as unknown as ListenerRes[]) || []
 
 	const showDisableReset = useSocketListener('show-disable-reset-points')
 	const showStoryPoints = showDisableReset?.message as unknown as boolean
 
-	// console.log('%c>>> CardsContainer: allUsersStoryPoints', 'color: yellow', allUsersStoryPoints)
-	// console.log('%c>>> CardsContainer: usersPointsData', 'color: yellow', usersPointsData)
-	// console.log('%c>>> CardsContainer: showDisableReset', 'color: #f50', showDisableReset)
+	// -99 = user joined room and hasn't participated yet.
+	// -55 = user has participated at least once and had points cleared.
+	const makeStringPoints = (message: number) => {
+		if (message === -99 || message === -55) return '--'
+		if (message === -1) return '?'
+		return message.toString()
+	}
 
-	// TODO: remove next line when development done
-	const sampleNames = namesArray.slice(10, 15)
+	const numberOfBlankCards = usersPointsData.filter(
+		({ message }) => message === -99 || message === -55,
+	).length
 
-	return (
-		<div className='py-2 flex justify-center items-center flex-wrap text-center gap-8 text-gray-300 border-0 border-gray-800'>
-			{usersPointsData.map(({ message, userName, timeStamp }, index) => {
-				const storyPoint = message === 'join' ? '-' : message.toString()
-				const imageNumber = index % sealifeImages.length
-				const imageSrc = sealifeImages[imageNumber]
-				return (
-					<UserCard
-						key={timeStamp}
-						name={userName}
-						storyPoint={storyPoint}
-						imageSrc={imageSrc}
-						showPoints={showStoryPoints}
-					/>
-				)
-			})}
+	// console.log('%c>>> CardsContainer: allUsersStoryPoints', 'color: #f70', allUsersStoryPoints)
+	// console.log('%c>>> CardsContainer: usersPointsData', 'color: red', usersPointsData)
+	// console.log('%c>>> CC socketListener: sortedUsersPoints', 'color: #5f0', sortedUsersPoints)
+	// console.log('%c>>> CardsContainer: showStoryPoints', 'color: #f60', showStoryPoints)
 
-			{/* TODO: FOR DEVELOPMENT ONLY - REMOVE WHEN DONE */}
-			{sampleNames.map((name, index) => {
-				const storyPoints = '00'
-				const imageNumber = (98 - index) % sealifeImages.length
-				const imageSrc = sealifeImages[imageNumber]
-				return (
-					<UserCard
-						key={name}
-						name={`${name} Test`}
-						storyPoints={storyPoints}
-						imageSrc={imageSrc}
-						showPoints={showStoryPoints}
-					/>
-				)
-			})}
+	return (
+		<div className='relative pt-2 pb-56 flex justify-center items-center flex-wrap text-center gap-8 text-gray-300 border-0 border-red-800 w-full'>
+			{showStoryPoints
```

Comment:
thought: this is super odd, i understand you are fighting motion but... meh. i'd leave a code comment explaining this was you explained in the video

Reply:
Okay, I'll leave a comment. I first coded it the way it is now, got it working, then said, "Okay, now make it one component and pass in different values." And it just wouldn't work. 

I think it needed something to tell motion to fire the animation when the array changes. I'm guessing the animation runs when the component mounts?? I'm sure there is some way in motion to make this happen, some "trigger" that can be pulled.

Yes, let's look at it later. I'll add a comment!

Reply:
I just asked Claude AI if framer motion would re-run an animation upon a change in state. And the answer is YES!! It said that when the `key` changes, the animation will run again. I had the key set to the timestamp of the user/points value in the array. 

I will make a combo component that has a variable key AND incorporates all the needed props, calculations, and JSX and see if it works!